portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -69,16 +76,30 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String comment = request.getParameter("text_input");
-    String name = request.getParameter("name_input");
-    if (comment == null || name == null) {
+    GoogleIdTokenVerifier verifier = new GoogleIdTokenVerifier.Builder(new NetHttpTransport(), JacksonFactory.getDefaultInstance())
+      .setAudience(Collections.singletonList("813751014340-ht5ugfu1pqj5a4gqq7rbvjk0sffu00it.apps.googleusercontent.com")).build();
+    GoogleIdToken idToken = null;     
+    String key = request.getParameter("idtoken");
+    
+    try {
+      idToken = verifier.verify(key);
+    } catch (Exception e) {
```

Comment:
Does this throw if verification fails?

Reply:
An exception is thrown if verification fails - I'll add a comment to make that clear.